src/public/media.ts

```diff
@@ -77,3 +92,268 @@ export function captureImage(callback: (error: SdkError, files: File[]) => void)
+
+  /**
+   * Additional properties for customization of select media in mobile devices
+   */
+  imageProps?: ImageProps;
+}
+
+/**
+ *  All properties in ImageProps are optional and have default values in the platform
+ */
+export interface ImageProps {
+  /**
+   * Optional; Lets the developer specify the image source, more than one can be specified.
+   * Default value is both camera and gallery
+   */
+  sources?: Source[];
+
+  /**
+   * Optional; Specify in which mode the camera will be opened.
+   * Default value is Photo
+   */
+  startMode?: Mode;
+
+  /**
+   * Optional; indicate if inking on the selected Image is allowed or not
+   * Default value is true
+   */
+  ink?: boolean;
+
+  /**
+   * Optional; indicate if user is allowed to move between front and back camera
+   * Default value is true
+   */
+  cameraSwitcher?: boolean;
+
+  /**
+   * Optional; indicate if putting text stickers on the selected Image is allowed or not
+   * Default value is true
+   */
+  textSticker?: boolean;
+
+  /**
+   * Optional; indicate if image filtering mode is enabled on the selected image
+   * Default value is false
+   */
+  enableFilter?: boolean;
+}
+
+/**
+ * The modes in which camera can be launched in select Media API
+ */
+export const enum Mode {
+  Photo = 1,
+  Document = 2,
+  Whiteboard = 3,
+  BusinessCard = 4,
+}
+
+/**
+ * Specifies the image source
+ */
+export const enum Source {
+  Camera = 1,
+  Gallery = 2,
+}
+
+/**
+ * Specifies the type of Media
+ */
+export const enum MediaType {
+  Image = 1,
+  Video = 2,
+  ImageOrVideo = 3,
+  Audio = 4,
+}
+
+/**
+ * Input for view images API
+ */
+export interface ImageUri {
+  value: string;
+  type: ImageUriType;
+}
+
+/**
+ * ID contains a mapping for content uri on platform's side, URL is generic
+ */
+export const enum ImageUriType {
+  ID = 1,
+  URL = 2,
+}
+
+/**
+ * Media chunks an output of getMedia API from platform
+ */
+export interface MediaChunk {
+  /**
+   * Base 64 data for the requested uri
+   */
+  chunk: string;
+
+  /**
+   * chunk sequence number​
+   */
+  chunkSequence: number;
+}
+
+/**
+ * Output of getMedia API from platform
+ */
+interface MediaResult {
+  /**
+   * error encountered in getMedia API
+   */
+  error: SdkError;
+
+  /**
+   * Media chunk which will be assemebled and converted into a blob
+   */
+  mediaChunk: MediaChunk;
+}
+
+/**
+ * Helper object to assembled media chunks
+ */
+export interface AssembleAttachment {
+  sequence: number;
+  file: Blob;
+}
+
+/**
+ * Helper class for assembling media
+ */
+interface MediaHelper {
+  mediaMimeType: string;
+  assembleAttachment: AssembleAttachment[];
+}
+
+/**
+ * Select an attachment using camera/gallery
+ * @param mediaInputs The input params to customize the media to be selected
+ * @param callback The callback to invoke after fetching the media
+ */
+export function selectMedia(mediaInputs: MediaInputs, callback: (error: SdkError, attachments: Media[]) => void): void {
```

Comment:
same here support success and failure separate callback.

Reply:
After we introduced SdkError we had decided we will keep error as the 1st param of callback and remaining subsequently. When we move to promises we can do something like: Promise.resolve([error, data]). Right now in case of error we return error and in case of success we return array of attachments.